### PR TITLE
add encoding/decoding cal::relativedelta

### PR DESCRIPTION
**Pull request by fmoor**

### CODE REVIEW

File: tests/test_datetime.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ async def test_duration_01(self):
         ''', durs)
         self.assertEqual(list(durs_from_db), durs)
 
-    @unittest.expectedFailure  # no relative delta in edgedb yet
+    @unittest.expectedFailure
```
Comment (fmoor) [inline tests/test_datetime.py:65]: There is a bit of a chicken and egg problem here. The server updates need to be published before these tests should be expected to pass, but the server needs the changes in this PR for it's own test suite.

File: edgedb/protocol/codecs/codecs.pyx [hunk 2]
```diff
@@ -581,5 +609,10 @@ cdef register_base_scalar_codecs():
         pgproto.jsonb_encode,
         pgproto.jsonb_decode)
 
+    register_base_scalar_codec(
+        'std::relativedelta',
```
Comment (elprans) [inline edgedb/protocol/codecs/codecs.pyx:613]:
```suggestion
        'cal::relativedelta',
```

File: edgedb/protocol/codecs/edb_types.pxi [hunk 3]
```diff
@@ -27,4 +27,5 @@ TYPE_IDS = {
     'std::duration': uuid.UUID('00000000-0000-0000-0000-00000000010e'),
     'std::json': uuid.UUID('00000000-0000-0000-0000-00000000010f'),
     'std::bigint': uuid.UUID('00000000-0000-0000-0000-000000000110'),
+    'std::relativedelta': uuid.UUID('00000000-0000-0000-0000-000000000111'),
```
Comment (elprans) [inline edgedb/protocol/codecs/edb_types.pxi:30]:
```suggestion
    'cal::relativedelta': uuid.UUID('00000000-0000-0000-0000-000000000111'),
```

**Review by elprans — APPROVED** [review]

**Review by elprans — CHANGES_REQUESTED** [review]
Actually, let's hold off merging this for now.

File: edgedb/protocol/codecs/codecs.pyx [hunk 4]
```diff
@@ -20,6 +20,7 @@
 import decimal
 import uuid
 import datetime
+from edgedb.datatypes.datatypes import RelativeDelta
```
Comment (1st1) [inline edgedb/protocol/codecs/codecs.pyx:23]: We usually import modules in python/cython code

Comment (1st1) [reply edgedb/protocol/codecs/codecs.pyx:23]:
```suggestion
from edgedb.datatypes import datatypes
```

File: edgedb/protocol/codecs/edb_types.pxi (hunk 3 above)
Comment (fmoor) [reply edgedb/protocol/codecs/edb_types.pxi:30]: Should I also change `std::local_datetime` etc. to `cal`?

Comment (elprans) [reply edgedb/protocol/codecs/edb_types.pxi:30]: Yes, that was an oversight.

File: edgedb/protocol/codecs/codecs.pyx (hunk 4 above)
Comment (fmoor) [reply edgedb/protocol/codecs/codecs.pyx]: changed

File: edgedb/protocol/codecs/edb_types.pxi (hunk 3 above)
Comment (fmoor) [reply edgedb/protocol/codecs/edb_types.pxi]: changed

File: edgedb/protocol/codecs/codecs.pyx (hunk 2 above)
Comment (fmoor) [reply edgedb/protocol/codecs/codecs.pyx]: changed

**Comment by fmoor** [comment]
> Actually, let's hold off merging this for now.

@elprans was this for the name change?

**Review by elprans — APPROVED** [review]
> @elprans was this for the name change?

Yes.  LGTM

**Pull request merged**